trunk-recorder/systems/smartnet_parser.cc

```diff
@@ -148,147 +176,320 @@ if (x.size()<3) {
+        // System ID
+        // BOOST_LOG_TRIVIAL(warning)
+        //     << "[" << system->get_short_name() << "] [Sys ID] SID $"
+        //     << std::hex << stack[3].full_address
+        //     << ", CC $" << std::hex << stack[2].full_address - 0x2800
+        //     << " -> " << getfreq(stack[2].full_address - 0x2800, system);
+        message.message_type = SYSID;
+        message.freq         = getfreq(stack[2].full_address - 0x2800, system);
+        message.sys_id       = stack[3].full_address;
+        messages.push_back(message);
+        return messages;
+      }
+    }
+    // further work needs to be done on patches/msels
+    if (stack[2].cmd == 0x340) {
+      // we have a patch
+      ++numConsumed;
       message.message_type = UNKNOWN;
+      messages.push_back(message);
+      return messages;
     }
-  } else  {
-    // Call continuation
-    if (groupflag) {
-      message.talkgroup    = full_address;
-      message.message_type = UPDATE;
-      cout << "UPDATE [ Freq: " << fixed << getfreq(command) << " TG: " << dec << address << " Full: " << dec << full_address << " CMD: " << hex << command << " Last CMD: " << hex << stack[1].cmd  << " ] Last CMD: " << hex << lastcmd << " GRoup: " << groupflag << endl;
+    if (stack[2].cmd == OSW_TY2_AFFILIATION) {
+      // we have an affiliation
+      ++numConsumed;
+      // BOOST_LOG_TRIVIAL(warning)
+      //     << "[" << system->get_short_name() << "] [affiliation] RID $"
+      //     << std::hex << stack[3].full_address
+      //     << " TG $" << std::hex << stack[2].full_address;
+      message.message_type = AFFILIATION;
+      message.talkgroup    = stack[2].full_address;
+      message.source       = stack[3].full_address;
+      messages.push_back(message);
+      return messages;
     }
   }
-}
 
-if (command == 0x03c0) {
-  message.message_type = STATUS;
 
-  // cout << "Status Command: " << hex << command << " Last CMD: 0x" <<  hex
-  // << lastcmd << " Freq: " << message.freq << " Talkgroup: " << dec <<
-  // address << " Last Address: " << dec << lastaddress<< endl;
 
-  // parse_status(command, address,groupflag);
-}
-*/
+  // If we get here, we don't know about this OCW (and/or a combination of it with others beside it).
+  // Error accordingly and log the stack.
 
-messages.push_back(message);
-return messages;
+  // If we just got started, then we also could just be looking at the tail of a known message.
+  // We preloaded the stack on a fresh start so there shouldn't be errors, but we have not
+  // emptied and reloaded the stack on a CC retune. Future request.
+
+  // There is also the possibility of missing OSWs and having incomplete messages.
+  // Adding the logic to test for this might be nice to have (test could be "if we got here,
+  // this OSW is is missing a header or other OSWs that comprise a valid message -
+  // test if we know this OSW command though, and if we do, discard the OSW and move on")
+  // BOOST_LOG_TRIVIAL(warning)
```

Comment:
Intentional for this to be commented out? L473-474 makes mention that we'll be logging, but not seeing stdout/boost-log being active.

Reply:
Thank you for taking a look @bctrainers. For the time being, I've commented out every logger. If a user wishes to do their own discovery and hunt for new and unknown OSWs, they may do so at their own peril. I'm hoping in the future when logging is standardized (#310) that this logging target be a better place to log everything.